src/humancellatlas/data/metadata/helpers/staging_area.py

```diff
@@ -0,0 +1,234 @@
+    """
+    A file describing one entity (e.g. biomaterial, protocol) in an subgraph.
+    """
+
+    def __init__(self, file_name: str, content: JSON):
+        assert file_name.endswith('.json')
+        file_id, version = file_name[:-5].split('_')
+        assert file_id == content['provenance']['document_id']
+        super().__init__(file_id, version, file_name, content)
+
+
+class DescriptorFile(File):
+    """
+    A file containing the checksums and other information for asserting the
+    integrity of a data file.
+    """
+
+    def __init__(self, file_name: str, content: JSON):
+        assert file_name.endswith('.json')
+        file_id, version = file_name[:-5].split('_')
+        super().__init__(file_id, version, file_name, content)
+
+    @property
+    def translated_content(self):
+        """
+        The content of a descriptor transformed into a format ready to create a
+        ManifestEntry object.
+        """
+        content = dict(self.content)
+        content['name'] = content.pop('file_name')
+        content['uuid'] = content.pop('file_id')
+        content['version'] = content.pop('file_version')
+        content['content-type'] = content.pop('content_type')
+        content['indexed'] = True
+        return content
+
+
+class StagingArea:
+
+    def __init__(self,
+                 repo_owner: str,
+                 repo_name: str,
+                 repo_branch: str,
+                 repo_path: Optional[str]
+                 ):
+        token = os.environ['GH_API_TOKEN']  # A GitHub personal access token
+        logger.debug('Requesting GitHub repo %s/%s', repo_owner, repo_name)
+        self.github_api = Github(token)
+        self.log_rate_limit()
+        self.repo = self.github_api.get_repo(f'{repo_owner}/{repo_name}')
+        branch = self.repo.get_branch(repo_branch)
+        self.branch_sha = branch.commit.sha
+        self.path = () if not repo_path else tuple(repo_path.split('/'))
+        # Fetch all the links, metadata, and descriptor files from the staging
+        # area and store the files in a mapping by their file ID.
+        self.links: Mapping[str, LinksFile]
+        self.metadata: Mapping[str, MetadataFile]
+        self.descriptors = Mapping[str, DescriptorFile]
+        self.links = self.get_files(self.path + ('links',))
+        self.metadata = self.get_files(self.path + ('metadata',))
+        self.descriptors = self.get_files(self.path + ('descriptors',))
+        self.log_rate_limit()
+
+    def log_rate_limit(self):
+        """
+        Log the current GitHub rate limit values
+        """
+        rate_limit = self.github_api.get_rate_limit()
+        logger.debug('GitHub API rate limit limit=%i, remaining=%i, reset=%s',
+                     rate_limit.core.limit, rate_limit.core.remaining, rate_limit.core.reset)
+
+    def get_files(self,
+                  path: Tuple[str, ...]
+                  ) -> MutableMapping[str, Union[LinksFile, MetadataFile, DescriptorFile]]:
```

Comment:
```suggestion
                  ) -> MutableMapping[str, JsonFile]:
```

and see comment about pulling up the `.json` assert